### DISCUSSION TITLE
Reusability of DnsNameResolver

### DISCUSSION THREAD

**Discussion by ceremo** [grpc/grpc-java#11586] [General]
Hi,

I have a question regarding gRPC's internal implementation of `DnsNameResolver` (`io.grpc.internal`). Netty also provides its own `DnsNameResolver` (`io.netty.resolver.dns`), and according to the changelog (https://github.com/netty/netty/compare/netty-4.1.113.Final...netty-4.1.114.Final), recent versions seem to be actively improving it. The question is why do you need to define your own implementation of `DnsNameResolver`?

Thanks for your time.

**Comment by ejona86** [comment]
Those name resolvers are completely different APIs. gRPC's DnsNameResolver mostly leverages libc's getaddrinfo; it's an adapter to gRPC's NameResolver/LoadBalancer infrastructure. Netty's DnsNameResolver is a full DNS client implementation.

We've considered having gRPC's DnsNameResolver use Netty's DnsNameResolver, but there's some technical details and we don't gain too much for typical gRPC workloads.
